src/rgw/rgw_common.h

```diff
@@ -1266,18 +1268,18 @@ struct RGWBucketInfo
      }
      swift_versioning = false;
      swift_ver_location.clear();
-     if (struct_v >= 16) {
+     swift_his_location.clear();
+     if (struct_v >= 18) {
```

Comment:
Something weird happened here. IIUC `swift_versioning` and `swift_ver_location` can be present since `struct_v` 16th but now we expect it only since 18th. I believe we shouldn't touch this but instead introduce:
```cpp
     if (struct_v >= 20 && swift_versioning) {
       ::decode(swift_his_location, bl);
     }
```
just before `DECODE_FINISH(bl)`.

Reply:
thanks, @rzarzynski !